Add new GET endpoint to retrieve all liquidated margin positions

godwinobas opened the pull request:
Added a new endpoint `GET /liquidated` to retrieve all liquidated margin positions. This endpoint returns a list of positions that have been liquidated (have a non-null `liquidated_at` timestamp).

djeck1432 commented on margin_app/app/api/margin_position.py:
```diff
@@ -65,3 +66,29 @@ async def close_margin_position(
+
+@router.get("/liquidated", response_model=List[MarginPositionResponse])
+async def get_all_liquidated_positions(
+    db: AsyncSession = Depends(margin_position_crud.session),
```
You don't need to use here Depends, as margin_position_crud has already all what you need

godwinobas replied:
Noted

djeck1432 reviewed (changes requested):
don't forget to add also to CRUD get_all_liquidated_positions method

godwinobas commented:
I've implemented the get_all_liquidated_positions endpoint with the following changes:

1. Added a new CRUD method in margin_position.py to fetch all liquidated positions
2. Created a new endpoint in margin_position.py that uses this CRUD method
3. Removed unnecessary Depends injection since margin_position_crud already manages the session
4. Added error handling and type hints

CBoYXD commented on margin_app/app/crud/margin_position.py:
```diff
@@ -48,4 +51,20 @@ async def close_margin_position(self, position_id : uuid.UUID) -> MarginPosition
             await self.write_to_db(position)
             return position.status
 
+    async def get_all_liquidated_positions(self) -> List[MarginPositionResponse]:
```
Use `get_objects` method from [`DBConnector`](https://github.com/djeck1432/spotnet/blob/main/margin_app/app/crud/base.py)

godwinobas replied:
Couldn't find a get_all method in DBConnector. Thanks for the pointer.

CBoYXD commented on margin_app/app/crud/margin_position.py:
```diff
@@ -48,4 +51,49 @@ async def close_margin_position(self, position_id : uuid.UUID) -> MarginPosition
             await self.write_to_db(position)
             return position.status
 
+    async def get_objects(
```
You don't need to implement this method, it already in `main` branch.

CBoYXD replied:
Add only the `where_clause` parameter, take everything else from main

CBoYXD reviewed (approved):
Almost good

djeck1432 commented:
@godwinobas Please, resolve merge conflict

CBoYXD reviewed (approved):
LGTM

djeck1432 reviewed (approved):
LGTM
